### ISSUE TITLE
Cannot parse ado in without any statements

### ISSUE THREAD

**Issue by f-f** [natefaubion/purescript-tidy#108]
The formatter is not happy about a file we have in our work codebase.
The file compiles fine with the latest compiler, but `purs-tidy` fails to parse it, and the error is just:
```
Unexpected '{'
```

The file is 4k lines long so it's hard to pinpoint exactly where the issue is, and it would be great if we could get a line number or some indication of where the parsing fails.

**Comment by f-f** [comment]
As an aside, @MonoidMusician found the culprit for the error, it's because of an empty `ado`:
```
ado
in { someField }
```

The CST parser doesn't seem to think that should be valid (and neither do I)

**Comment by natefaubion** [comment]
I agree, and I think an empty `ado/in` is pretty silly, but the compiler allows it and it desugars to `pure`.

**Comment by MonoidMusician** [comment]
Minimal repro:
```purescript
module MuchAdoAboutNothing where

much = ado
  in Nothing

data Maybe a = Just a | Nothing

pure a = a
```
gives `Unexpected end of indented block` but compiles fine with `purs compile`

To see the specific error from OP, you need this:
```purescript
a = much where
  much {} = ado
    in Nothing
```

In particular, it needs to be in a `where` and the error is targeting the function argument (`much y = …` gives `Unexpected identifier y`, nothing else is valid there). But that's just funky errors getting pushed backwards due to backtracking :)

I can put up a PR soon

**Comment by natefaubion** [comment]
This was addressed in the 0.9.3 release (along with printing locations for errors).
